File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;
        vector<int> h(26);
        for(int i=0; i<s.length(); i++){
            h[s[i] - 'a']++;
            h[t[i] - 'a']--;
        }
        for(auto &it:h){
            if (it != 0) return false;
        }
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) { return false }

    freq := make([]int, 26)
    for i := range s {
        s_idx := int(s[i] - 'a')
        t_idx := int(t[i] - 'a')

        freq[s_idx]++
        freq[t_idx]--
    }

    for _, f := range freq {
        if f != 0 { return false }
    }

    return true
}